### ISSUE TITLE
[RL] Loss function refactor in RL loop

### ISSUE THREAD

**Issue by wwwjn** [pytorch/torchtitan#2692]
In current RL loop, we used a single function [`compute_gradient_loss()`](https://github.com/pytorch/torchtitan/blob/8625f2248cbe360bd65307f0e16941b18b161bb3/torchtitan/experiments/rl/actors/utils.py#L62) to run the trainer model forward and compute loss. Will need a refactor to have a better separation of different functionality, and also easier to switch to any other loss function.

**Comment by tianyu-l** [comment]
seems done
